ident/ident.go

```diff
@@ -0,0 +1,75 @@
+package ident
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"sort"
+)
+
+type Identifiable interface {
+	Identity() []byte
+}
+
+func ContentAddress(entity Identifiable) string {
+	h := sha256.New()
+	_, _ = h.Write(entity.Identity())
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+type Buffer struct {
+	buf []byte
```

Comment:
Not sure why we ever want an actual `Buffer` for this, it could be quite long and fragment our memory.  If I'm only going to call `ContentAddress` on it, why not stream the data into the hash in the first place?

Reply:
I guess you're right. changed.